web-client/src/components/AddSockForm.tsx

```diff
@@ -0,0 +1,266 @@
+                                        return occurrences <= 1 || "Size must be unique";
+                                    },
+                                })}
+                                defaultValue={field.size || ""}
+                                className="border p-2 text-black text-sm"
+                            >
+                                <option value="" disabled>
+                                    Select Size
+                                </option>
+                                {availableSizes.map((size) => (
+                                    <option
+                                        key={size}
+                                        value={size}
+                                        disabled={
+                                            selectedSizes.includes(size) && field.size !== size
+                                        }
+                                    >
+                                        {size}
+                                    </option>
+                                ))}
+                            </select>
+
+                            <Input
+                                {...register(`sizes.${index}.quantity`, {
+                                    required: "Quantity is required",
+                                    valueAsNumber: true,
+                                    min: 1,
+                                })}
+                                placeholder="Quantity"
+                                type="number"
+                                className="border p-2 text-black"
+                            />
+
+                            <Input
+                                {...register(`sizes.${index}.price`, {
+                                    required: "Price is required",
+                                    valueAsNumber: true,
+                                    min: 0.01,
+                                })}
+                                placeholder="Price"
+                                type="number"
+                                className="border p-2 text-black"
+                            />
+
+                            <Button
+                                size="icon"
+                                variant="destructive"
+                                onClick={() => remove(index)}
+                                className="h-8 w-8 flex items-center justify-center bg-red-500 border-red-500 text-white"
+                            >
+                                <Trash2 className="h-4 w-4" />
+                            </Button>
+                        </div>
+                    ))}
+
+                    <Button
+                        type="button"
+                        variant="outline"
+                        className="w-full border-black text-black"
+                        onClick={() => append({ size: "S", quantity: 1, price: 0 })}
```

Comment:
Currently, you're always appending a size of "S" whenever the button is clicked. This should really add whatever the "next available size is". The easiest way to do this, is by adding an array to store the "nextSizesToSelect" (a list of sizes which haven't been selected) like this:

```ts
const selectedSizes = watch("sizes").map((item) => item.size);
const nextSizesToSelect = availableSizes.filter(
  (size) => !selectedSizes.includes(size),
);
```

Then, you can make the button be disabled when there are no sizes to select from. The overall button should look like this:

```ts
<Button
  type="button"
  variant="outline"
  className="w-full"
  onClick={() => {
    if (nextSizesToSelect.length > 0) {
      append({ size: nextSizesToSelect[0], quantity: 1, price: 0 });
    } else {
      console.error("Unable to add another size, no sizes available.")
    }
  }}
  disabled={nextSizesToSelect.length < 1}
>
  <Plus className="mr-2 h-4 w-4" /> Add Size
</Button>

```